galsim/nfw_halo.py

```diff
@@ -195,91 +184,66 @@ def __omega(self, a):
         """
         return self.cosmo.omega_m/(self.cosmo.E(a)**2 * a**3)
 
-    def __farcth (self, x, out=None):
+    def __farcth (self, x):
         """Numerical implementation of integral functions of a spherical NFW profile.
 
         All expressions are a function of `x`, which is the radius r in units of the NFW scale
         radius, r_s.  For the derivation of these functions, see for example Wright & Brainerd
         (2000, ApJ, 534, 34).
         """
-        if out is None:
-            out = np.zeros_like(x)
+        out = np.zeros_like(x, dtype=float)
 
         # 3 cases: x > 1, x < 1, and |x-1| < 0.001
-        mask = (x < 0.999)
-        if mask.any():
-            a = ((1.-x[mask])/(x[mask]+1.))**0.5
-            out[mask] = 0.5*np.log((1.+a)/(1.-a))/(1-x[mask]**2)**0.5
+        mask = np.where(x < 0.999)[0]
```

Comment:
Why `np.where` (Here and below)?  I think the original is easier to read.  (Don't have to remember why there's a `[0]`).  Getting rid of the `mask.any()` is a definite improvement though.

Reply:
The reason for this is efficiency.  Indexing with an array of bools tends to be slower than indexing with an array of integers saying where the bools were True.  Especially if there are far fewer True than False, which I suppose will be the case for typically 2 out of these 3 cases.